src/sentry/tasks/post_process.py

```diff
@@ -1016,7 +1017,7 @@ def process_code_mappings(job: PostProcessJob) -> None:
         else:
             return
 
-        derive_code_mappings.delay(project.id, event.data)
+        auto_source_code_config.delay(project.id, event.data)
```

Comment:
You could drop task activations here. During a deploy workers are replaced incrementally, and a 'new' worker could  spawn this task, which is processed by an 'old' worker. The old worker won't have the new task definition and the activation will be lost.

What I've done in the past for this kind of situation is use an option to choose which task is scheduled. Once all workers are restart, the option can be flipped and the new task will be scheduled. This helps clean up to be done in one pull request and deploy instead of needing to use two or three.